src/Analytics.js

```diff
@@ -147,6 +148,23 @@ class Analytics {
         return true;
     }
 
+    startPageChangeTimer() {
+        performance.clearMarks('riot_page_change_start');
+        performance.mark('riot_page_change_start');
+    }
+
+    stopPageChangeTimer() {
+        performance.mark('riot_page_change_stop');
+        performance.measure(
+            'riot_page_change_delta',
+            'riot_page_change_start',
+            'riot_page_change_stop',
+        );
+
+        const measurement = performance.getEntriesByName('riot_page_change_delta').pop();
+        this.generationTimeMs = measurement.duration;
```

Comment:
The way you're calling this (ie. with willupdate / didupdate) this looks fine, but if these can ever overlap then this will go very wonky, so might be worth calling this out. Also, any reason, not to clear the marks as soon as we have our number?

Reply:
>might be worth calling this out

As in, with docs on the functions?

>Also, any reason, not to clear the marks as soon as we have our number?

Nope, I guess not.

Reply:
Yeah, or in conjunction with comment below, change the API so start returns some unique handle that you then pass into `trackPageChange`? (and remove `stopPageChangeTimer`). It may be more effort than it's worth though.